server/controllers/authController.js

```diff
@@ -0,0 +1,26 @@
+import { Router } from "express";
+import { authService } from '../services';
+import { catchAll } from '../utilities';
+
+const login = async (req, res) => {
+    if (!req.body || !req.body.username || !req.body.password) {
+        throw new TypeError('Missing params!')
+    }
+
+    const username = req.body.username;
+    const password = req.body.password;
+
+    const response = await authService.handleLogin(username, password);
+
+    res.json({ success: true, ...response});
+}
+
+// instantiate new router
+const authController = new Router();
+
+// bind routes to functions
+authController.post('/login', catchAll(login));
```

Comment:
This `catchAll` function is nice! It would be even better if it offloaded to a global error handler rather than calling `next`. The reason for this, is that the default for the server is return HTML, and errors are much more useful as `json`. If you want do make this change, I can help design the error handler just lemme know!